algorithms/src/fft/polynomial/mod.rs

```diff
@@ -30,21 +31,69 @@ mod sparse;
 pub use sparse::SparsePolynomial;
 
 /// Represents either a sparse polynomial or a dense one.
-#[derive(Clone)]
-pub enum DenseOrSparsePolynomial<'a, F: 'a + Field> {
+#[derive(Clone, Debug)]
+pub enum DenseOrSparsePolynomial<'a, F: Field> {
     /// Represents the case where `self` is a sparse polynomial
     SPolynomial(Cow<'a, SparsePolynomial<F>>),
     /// Represents the case where `self` is a dense polynomial
     DPolynomial(Cow<'a, DensePolynomial<F>>),
 }
```

Comment:
More of a broad design question, why not refactor the enum `DenseOrSparsePolynomial` as:
```rs
pub enum Polynomial {
    Sparse(..),
    Dense(..),
}
```
so that in match cases we have:
```rs
match polynomial {
    Polynomial::Sparse(..) => (),
    Polynomial::Dense(..) => (),
}

Reply:
Yeah, I think we should do that. I've certainly been encountering the verbosity in my refactor rn.

Reply:
See https://github.com/AleoHQ/snarkVM/issues/656